ironfish/src/wallet/wallet.ts

```diff
@@ -680,6 +681,28 @@ export class Accounts {
     try {
       this.assertHasAccount(sender)
 
+      // check if the chain data is fully synced
+      let headHash = this.headHashes.get(sender.id)
+      headHash = await sender.getHeadHash()
+      if (!headHash) {
```

Comment:
Is there a reason we check twice? Does `sender.getHeadHash()` do something that would change the result of checking the first time?